### ISSUE TITLE
ros2_overlay_ws colcon build --symlink-install error

### ISSUE THREAD

**Issue by arun0303** [intel/ros2_openvino_toolkit#155]
During step 4. Building and Installation
-->Install ROS2_OpenVINO packages
-->Build package
cd ~/ros2_overlay_ws
colcon build --symlink-install
 The following error 
make[2]: *** No rule to make target '/usr/lib/x86_64-linux-gnu/libboost_python3.so', needed by 'src/libcv_bridge.so'.  Stop.
make[2]: *** Waiting for unfinished jobs....
make[1]: *** [src/CMakeFiles/cv_bridge.dir/all] Error 2
make: *** [all] Error 2

**Comment by portgasray** [comment]
> During step 4. Building and Installation
> -->Install ROS2_OpenVINO packages
> -->Build package
> cd ~/ros2_overlay_ws
> colcon build --symlink-install
> 
> ## [Processing: cv_bridge, message_filters, object_msgs, realsense_msgs]
> --- stderr: cv_bridge
> make[2]: *** No rule to make target '/usr/lib/x86_64-linux-gnu/libboost_python3.so', needed by 'src/libcv_bridge.so'. Stop.
> make[2]: *** Waiting for unfinished jobs....
> make[1]: *** [src/CMakeFiles/cv_bridge.dir/all] Error 2
> make: *** [all] Error 2
> Failed <<< cv_bridge [ Exited with code 2 ]
> Aborted <<< realsense_msgs
> Aborted <<< message_filters
> Aborted <<< object_msgs
> 
> Summary: 0 packages finished [2min 4s]
> 1 package failed: cv_bridge
> 3 packages aborted: message_filters object_msgs realsense_msgs
> 1 package had stderr output: cv_bridge
> 15 packages not processed

u should go to the repo [ros2_intel_realsense](https://github.com/intel/ros2_intel_realsense), and see the README, to install the dependencies, such as
```
sudo apt-get install ros-dashing-cv-bridge ros-dashing-librealsense2 ros-dashing-message-filters ros-dashing-image-transport
```

Check that! Good Luck!

**Comment by arun0303** [comment]
> > During step 4. Building and Installation
> > -->Install ROS2_OpenVINO packages
> > -->Build package
> > cd ~/ros2_overlay_ws
> > colcon build --symlink-install
> > ## [Processing: cv_bridge, message_filters, object_msgs, realsense_msgs]
> > --- stderr: cv_bridge
> > make[2]: *** No rule to make target '/usr/lib/x86_64-linux-gnu/libboost_python3.so', needed by 'src/libcv_bridge.so'. Stop.
> > make[2]: *** Waiting for unfinished jobs....
> > make[1]: *** [src/CMakeFiles/cv_bridge.dir/all] Error 2
> > make: *** [all] Error 2
> > Failed <<< cv_bridge [ Exited with code 2 ]
> > Aborted <<< realsense_msgs
> > Aborted <<< message_filters
> > Aborted <<< object_msgs
> > Summary: 0 packages finished [2min 4s]
> > 1 package failed: cv_bridge
> > 3 packages aborted: message_filters object_msgs realsense_msgs
> > 1 package had stderr output: cv_bridge
> > 15 packages not processed
> 
> u should go to the repo [ros2_intel_realsense](https://github.com/intel/ros2_intel_realsense), and see the README, to install the dependencies, such as
> 
> ```
> sudo apt-get install ros-dashing-cv-bridge ros-dashing-librealsense2 ros-dashing-message-filters ros-dashing-image-transport
> ```
> 
> Check that! Good Luck!

Yeah those dependencies were installed previously and tried again also but no effect. pls see the screenshot.
![Screenshot from 2019-11-25 11-32-03](https://user-images.githubusercontent.com/41463803/69516305-4c121600-0f77-11ea-9d38-8d1b41df250b.png)

**Comment by portgasray** [comment]
> > > During step 4. Building and Installation
> > > -->Install ROS2_OpenVINO packages
> > > -->Build package
> > > cd ~/ros2_overlay_ws
> > > colcon build --symlink-install
> > > ## [Processing: cv_bridge, message_filters, object_msgs, realsense_msgs]
> > > --- stderr: cv_bridge
> > > make[2]: *** No rule to make target '/usr/lib/x86_64-linux-gnu/libboost_python3.so', needed by 'src/libcv_bridge.so'. Stop.
> > > make[2]: *** Waiting for unfinished jobs....
> > > make[1]: *** [src/CMakeFiles/cv_bridge.dir/all] Error 2
> > > make: *** [all] Error 2
> > > Failed <<< cv_bridge [ Exited with code 2 ]
> > > Aborted <<< realsense_msgs
> > > Aborted <<< message_filters
> > > Aborted <<< object_msgs
> > > Summary: 0 packages finished [2min 4s]
> > > 1 package failed: cv_bridge
> > > 3 packages aborted: message_filters object_msgs realsense_msgs
> > > 1 package had stderr output: cv_bridge
> > > 15 packages not processed
> > 
> > 
> > u should go to the repo [ros2_intel_realsense](https://github.com/intel/ros2_intel_realsense), and see the README, to install the dependencies, such as
> > ```
> > sudo apt-get install ros-dashing-cv-bridge ros-dashing-librealsense2 ros-dashing-message-filters ros-dashing-image-transport
> > ```
> > 
> > 
> > Check that! Good Luck!
> 
> Yeah those dependencies were installed previously and tried again also but no effect. pls see the screenshot.
> ![Screenshot from 2019-11-25 11-32-03](https://user-images.githubusercontent.com/41463803/69516305-4c121600-0f77-11ea-9d38-8d1b41df250b.png)

`sudo apt remove ros-dashing-librealsense2`, give it a shot! (I am not sure if this will work!)

then I will use `rm install log build -rf` and `colcon build --symlink-install` to reproduce the error under _ros2 workspace_.

**Comment by arun0303** [comment]
> `sudo apt remove ros-dashing-librealsense2`, give it a shot! (I am not sure if this will work!)
> 
> then I will use `rm install log build -rf` and `colcon build --symlink-install` to reproduce the error under _ros2 workspace_.

It did not worked.
![Screenshot from 2019-11-28 15-32-12](https://user-images.githubusercontent.com/41463803/69796636-5c82f480-11f4-11ea-90aa-8997f9a41b5c.png)

**Comment by portgasray** [comment]
> > `sudo apt remove ros-dashing-librealsense2`, give it a shot! (I am not sure if this will work!)
> > then I will use `rm install log build -rf` and `colcon build --symlink-install` to reproduce the error under _ros2 workspace_.
> 
> It did not worked.

since your error caused by boost, check whether installed correctly!

**Comment by arun0303** [comment]
> > > `sudo apt remove ros-dashing-librealsense2`, give it a shot! (I am not sure if this will work!)
> > > then I will use `rm install log build -rf` and `colcon build --symlink-install` to reproduce the error under _ros2 workspace_.
> > 
> > 
> > It did not worked.
> 
> since your error caused by boost, check whether installed correctly!
I had verified boost is installed properly. pls find the screenshot
![Screenshot from 2019-12-02 15-29-53](https://user-images.githubusercontent.com/41463803/69950064-b21f1000-1518-11ea-8c15-97f5e9f0e933.png)

**Comment by portgasray** [comment]
see the error information: `needed by 'src/libcv_bridge.so'. Stop.` 
there is no` libcv_bridge.so` in my `ros2_overlay_ws/src` , so if you can show me where the `libcv_bridge.so` file

**Comment by arun0303** [comment]
libcv_bridge.so  file is not available in my ros2_overlay_ws/src directory and its available inside /opt/ros/dashing/lib directory.

**Comment by portgasray** [comment]
> libcv_bridge.so file is not available in my ros2_overlay_ws/src directory and its available inside /opt/ros/dashing/lib directory.

more information about the `ros2_overlay_ws/src/vision_opencv`:
* git branch

**Comment by portgasray** [comment]
yeah, you should check the CMakeLists under `cv_bridge`, you'll find the valid information: 
```
      # This is a bit of a hack to suppress a warning
      #   No header defined for python3; skipping header check
      # Which should only affect Boost versions < 1.67
      # Resolution for newer versions:
      #  https://gitlab.kitware.com/cmake/cmake/issues/16391
```
the sub-links in the [issuse](https://gitlab.kitware.com/cmake/cmake/issues/16391) are very useful, especially the last one, 
![_20191203135920](https://user-images.githubusercontent.com/17681580/70024535-61212180-15d5-11ea-98c8-7557d4fe9fb0.png)

you should view it carefully and  according to my experience, try the following:
update [line30 ](https://github.com/ros-perception/vision_opencv/blob/b19d3d10c0c822c454047c1c4950ab22dc4a9b93/cv_bridge/CMakeLists.txt#L30) in `ros2_overlay_ws/src/vision_opencv/cv_bridge`
![image](https://user-images.githubusercontent.com/17681580/70024540-67af9900-15d5-11ea-9637-7c37ab9bbb46.png)

**Comment by hariharan382** [comment]
please anyone help me to solve the error 
error while running"colcon build --symlink-install"

**Comment by jb-balaji** [comment]
Stale issue, hence closed, kindly create new issue if required.
